src/ensembl/src/content/app/browser/zmenu/Zmenu.test.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+import faker from 'faker';
+
+import Zmenu, { ZmenuProps, Tip } from './Zmenu';
+import ZmenuContent from './ZmenuContent';
+
+import { createZmenuContent } from 'tests/fixtures/browser';
+import configureStore from 'src/store';
+
+describe('<Zmenu />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const store = configureStore();
+  const defaultProps: ZmenuProps = {
+    anchor_coordinates: {
+      x: 490,
+      y: 80
+    },
+    browserRef: {
+      current: document.createElement('div')
+    },
+    content: createZmenuContent(),
+    id: faker.lorem.words(),
+    onEnter: jest.fn(),
+    onLeave: jest.fn()
+  };
+
+  const wrappingComponent = (props: any) => (
+    <Provider store={store}>{props.children}</Provider>
+  );
+
+  let wrapper: any;
+
+  beforeEach(() => {
+    wrapper = mount(<Zmenu {...defaultProps} />, { wrappingComponent });
+  });
+
+  describe('rendering', () => {
+    test('renders zmenu content', () => {
+      expect(wrapper.find(ZmenuContent)).toHaveLength(1);
+    });
+
+    test('renders zmenu tip', () => {
+      expect(wrapper.find(Tip)).toHaveLength(1);
+    });
```

Comment:
Rendering of the tip is definitely not an interesting test 😄

Reply:
It's one of those tests that I really wasn't sure whether to add or not? 😄 Visually it should be present and something that can be missed out, however I get what you are implying; it is more about UI testing than testing the code.